src/NATS.Client.Core/INatsSerialize.cs

```diff
@@ -613,6 +618,11 @@ public void Serialize(IBufferWriter<byte> bufferWriter, T value)
     /// <inheritdoc />
     public T? Deserialize(in ReadOnlySequence<byte> buffer)
     {
+        if (buffer.Length == 0)
+        {
+            return default;
+        }
```

Comment:
Shouldn't this only be run if `T` is one of those that is normally handled by this deserailizer?  Otherwise it is not possible to chain a deserailzer that handles an empty payload for another `T`.